routers/user.controller.js

```diff
@@ -85,14 +89,44 @@ class UserController {
       }).validate(req.body);
 
       if (validationError) {
-        throw new ApiError(400, 'Bad requiest', validationError);
+        throw new ApiError(400, 'Bad request', validationError);
       }
 
       next();
     } catch (e) {
       errorHandler(req, res, e);
     }
   }
+
+  //========================================
+
+  async userLogout(req, res) {
+    try {
+      const { activeToken, user } = req;
+
+      user.tokens = user.tokens.filter(
```

Comment:
We've talked about `$pull` update operation for solving cases like that

Reply:
But will it delete whole object { token, expires}?

Reply:
The same thing `.filter` does, doesn't it?